gradio/routes.py

```diff
@@ -222,15 +223,17 @@ def login(form_data: OAuth2PasswordRequestForm = Depends()):
         def main(request: fastapi.Request, user: str = Depends(get_current_user)):
             mimetypes.add_type("application/javascript", ".js")
             blocks = app.get_blocks()
+            root_path = request.scope.get("root_path")
```

Comment:
```suggestion
            root_path = request.scope.get("root_path", "")
```
I don't believe root_path is guaranteed anywhere to be present, and we need to fall back on an empty string